spring-restdocs-asciidoctor/src/test/java/org/springframework/restdocs/asciidoctor/RestDocsIncludeBlockMacroTest.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright 2014-2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.restdocs.asciidoctor;
+
+import java.io.File;
+
+import org.asciidoctor.Asciidoctor;
+import org.asciidoctor.Attributes;
+import org.asciidoctor.Options;
+import org.hamcrest.CoreMatchers;
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Test;
+
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.CoreMatchers.startsWith;
+import static org.junit.Assert.assertThat;
+
+/**
+ * Tests for ruby based rest docs block macro
+ * Because there is no java implementation (yet)
+ * we can only test the behaviour when rendering
+ *
+ * @author Gerrit Meier
+ */
+public class RestDocsIncludeBlockMacroTest {
+
+	private final Options options = new Options();
+	private final Asciidoctor asciidoctor = Asciidoctor.Factory.create();
+
+	@Before
+	public void setUp() {
+		this.options.setAttributes(getAttributes());
+	}
+
+	private Attributes getAttributes() {
+		Attributes attributes = new Attributes();
+		attributes.setAttribute("projectdir", new File(".").getAbsolutePath());
+		return attributes;
+	}
+
+	@Test
+	public void includeSnippetWithLevel() {
+		String expectedContent = "<div class=\"sect1\">\n" +
```

Comment:
My preference would be to store the expected content in files in `src/test/resources`